markdown/developers/modules.md

```diff
@@ -696,6 +696,17 @@ ch_genome_bam.map {
 
 As you can see the `meta map` is a quite flexible way for storing meta data in channels. Feel free to add whatever other key-value pairs your pipeline may need to it. We're looking to add [Custom objects](https://github.com/nf-core/modules/issues/1338) which will lock down the usage a bit more.
 
+## What are the ext properties/keys?
+
+Ext properties or keys can be used to override module options. The following table lists the available keys.
+
+| Key        | Description                                                                 |
+| ---------- | --------------------------------------------------------------------------- |
+| ext.args   | Additional arguments appended to command in module.                         |
+| ext.args2  | Second set of arguments appended to command in module (multi-tool modules). |
+| ext.args3  | Third set of arguments appended to command in module (multi-tool modules).  |
+| ext.prefix | File name prefix for output files.                                          |
```

Comment:
Perhaps also add that this defaults to `meta.id`?

Reply:
I think it would be good to add more detail like the section above it. 
Perhaps link to the `ext` process directive: https://www.nextflow.io/docs/latest/process.html#ext
Explain that this directive is used to inject strings into the script.
Provide some examples of how this is used . 
* Example of setting args based on parameter settings: https://github.com/nf-core/rnaseq/blob/e049f51f0214b2aef7624b9dd496a404a7c34d14/conf/modules.config#L222-L226
* Setting prefix based on task inputs: https://github.com/nf-core/rnaseq/blob/e049f51f0214b2aef7624b9dd496a404a7c34d14/conf/modules.config#L297
* Example of setting args based on both parameters and task inputs: https://github.com/nf-core/rnaseq/blob/e049f51f0214b2aef7624b9dd496a404a7c34d14/conf/modules.config#L377-L381

Reply:
We may need to remove "(multi-tool modules)" because `ext.args2` can also be used for single-tool modules that need arguments in two positions, e.g. https://github.com/nf-core/modules/pull/2286#issuecomment-1277315963

Reply:
My goal here was really to include this info as Phil Ewels requested, and get things started. I didn't expect to complete the section entirely. Regarding the comment by @muffato, should "(multi-tool modules)" be removed? Can I get a confirmation on this?

Reply:
Sounds good. Yes, go ahead and remove the multi-tools. I'll fill in the extra stuff at a later date then.